platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +249,196 @@ def get_sfp(self, index):
+            used to pass a description of the reboot cause.
+            Some more causes are appended to the existing list to handle other
+            modules such as DPUs.
+            Ex: REBOOT_CAUSE_POWER_LOSS, REBOOT_CAUSE_HOST_RESET_DPU,
+            REBOOT_CAUSE_HOST_POWERCYCLED_DPU, REBOOT_CAUSE_SW_THERMAL,
+            REBOOT_CAUSE_DPU_SELF_REBOOT
+        """
+        if utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_from_main_board') == 1:
+            return self.REBOOT_CAUSE_HOST_RESET_DPU, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_dpu_thermal') == 1:
+            return self.REBOOT_CAUSE_SW_THERMAL, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_aux_pwr_or_reload') == 1:
+            return self.REBOOT_CAUSE_POWER_LOSS, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/reset_pwr_off') == 1:
+            return self.REBOOT_CAUSE_DPU_SELF_REBOOT, ''
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/tpm_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'Reset by the TPM module'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/perst_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'PERST# signal to ASIC'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/phy_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'Phy reset'
+        elif utils.read_int_from_file(f'/run/hw-management/system/dpu{self.dpu_id}/system/usbphy_rst') == 1:
+            return self.REBOOT_CAUSE_HARDWARE_OTHER, 'USB Phy reset'
+        else:
+            return self.REBOOT_CAUSE_NON_HARDWARE, ''
+
+    ##############################################
+    # Midplane methods for modular chassis
+    ##############################################
+    def get_midplane_ip(self):
+        """
+        Retrieves the midplane IP-address of the module in a modular chassis
+        When called from the Supervisor, the module could represent the
+        line-card and return the midplane IP-address of the line-card.
+        When called from the line-card, the module will represent the
+        Supervisor and return its midplane IP-address.
+        When called from the DPU, returns the midplane IP-address of the dpu-card.
+        When called from the Switch returns the midplane IP-address of Switch.
+
+        Returns:
+            A string, the IP-address of the module reachable over the midplane
+
+        """
+        key = "DHCP_SERVER_IPV4_PORT|" + "bridge-midplane" + "|" + self.get_name().lower()
+        data_dict = DpuModule.config_db.get_all(DpuModule.config_db.CONFIG_DB, key)
+        return data_dict["ips"]
+
+    def is_midplane_reachable(self):
```

Comment:
This should be part of the base vendor independent implementaion

Reply:
Removed